benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,736 @@
+    
+    return node_offsets, node_offsets_replicated, count_replicated
+
+def generate_rmat_dataset(dataset, seed=62, labeled_percentage=0.01, num_labels=256, reverse_edges=False, persist=False, add_edge_types=False):
+    """
+    Generates an rmat dataset.  Currently does not support heterogeneous datasets.
+
+    Parameters
+    ----------
+    dataset: The specifier of the rmat dataset (i.e. rmat_20_16)
+    seed: The seed to use for random number generation
+    num_labels: The number of classes for the labeled nodes
+    reverse_edges: Whether to reverse the edges in the edgelist (should be True for DGL, False, for PyG)
+    """
+
+    dataset = dataset.split('_')
+    scale = int(dataset[1])
+    edgefactor = int(dataset[2])
+
+    dask_edgelist_df = generate_edgelist_rmat(
+        scale=scale, edgefactor=edgefactor, seed=seed, unweighted=True, mg=True,
+    )
+    dask_edgelist_df = dask_edgelist_df.astype("int64")
+    dask_edgelist_df = dask_edgelist_df.reset_index(drop=True)
+
+
+    dask_edgelist_df = renumber_ddf(dask_edgelist_df).persist()
+    if persist:
+        dask_edgelist_df = dask_edgelist_df.persist()
+
+    dask_edgelist_df = symmetrize_ddf(dask_edgelist_df).persist()
+    if persist:
+        dask_edgelist_df = dask_edgelist_df.persist()
+
+    if add_edge_types:
+        dask_edgelist_df['etp'] = cupy.int32(0) # doesn't matter what the value is, really
+    
+    # generator = np.random.default_rng(seed=seed)
+    num_labeled_nodes = int(2**(scale+1) * labeled_percentage)
+    label_df = pd.DataFrame({
+        'node': np.arange(num_labeled_nodes),
+        # 'label': generator.integers(0, num_labels - 1, num_labeled_nodes).astype('float32')
+    })
+    
+    n_workers = len(default_client().scheduler_info()['workers'])
+    dask_label_df = ddf.from_pandas(label_df, npartitions=n_workers*2)
+    del label_df
+    gc.collect()
+
+    dask_label_df = dask_cudf.from_dask_dataframe(dask_label_df)
+
+    node_offsets = {'paper': 0}
+    edge_offsets = {('paper','cites','paper'):0}
+    total_num_nodes = int(dask_cudf.concat([dask_edgelist_df.src, dask_edgelist_df.dst]).nunique().compute())
+
+    if reverse_edges:
+        dask_edgelist_df = dask_edgelist_df.rename(columns={'src':'dst', 'dst':'src'})
+
+    return dask_edgelist_df, dask_label_df, node_offsets, edge_offsets, total_num_nodes
+
+
+def load_disk_dataset(dataset, dataset_dir='.', reverse_edges=True, replication_factor=1, persist=False, add_edge_types=False):
+    from pathlib import Path
+    path = Path(dataset_dir) / dataset
+    parquet_path = path / 'parquet'
+
+    with open(os.path.join(path, 'meta.json')) as meta_file:
+        meta = json.load(meta_file)
+    
+    node_offsets, node_offsets_replicated, total_num_nodes = \
+        assign_offsets_pyg(meta['num_nodes'], replication_factor=replication_factor)
+
+    edge_index_dict = {}
+    for edge_type in os.listdir(parquet_path):
+        if re.match(r'[a-z]+__[a-z]+__[a-z]+', edge_type):
```

Comment:
Minor nitpick: I would suggest using meta.json and ['num_edges'] for this rather than relying on regex matching.

Reply:
done